### ISSUE TITLE
Suggestion: I think the next update should utilize masteries instead of prestiging (1.6 spoilers)

### ISSUE THREAD

**Issue by argon932** [daleao/modular-overhaul#189]
1.6 adds mastery points, which you start getting after reaching level 10 with all skills. It doesn't make much sense to prestige now that we have the cool new mastery system. It would be awesome if the mod utilized this in the next update.

My suggestion of progress would be:
- After reaching level 10 with all skills, you begin gaining mastery points as normal.
- After earning a mastery point, the first unlock of each profession will be the vanilla mastery unlock.
- After that, if you use an additional mastery point on a profession again, you simply get to choose one of the perks you don't have yet, resetting your mastery bar instead of your profession level.
- After unlocking all vanilla masteries and all profession perks (15 mastery points). However, you still gain mastery points.
- Spending these additional mastery points would allow you to improve various things depending what profession you choose. Each profession would have a cap of 20 levels (100 total mastery points to fully max out). This would help offset the fact that we wouldn't be leveling skills to level 20 anymore. 

Some examples.

Combat:
+2% Damage
+1% Crit Chance
+2% HP

Fishing:
+2% Catch Speed
+2% Fishing Bar Size
+5% Fish sell price

Farming:
+1% chance of harvested crops being highest quality
+2% Artisan goods / Animal products / Crops sell price

Foraging:
+ 10% chance for additional wood / hardwood when cutting trees
+ 5% forage/syrup sell price
+ 3% tapper speed
+ 3% chance at additional forage when harvesting

Mining:
10% chance for additional stone when mining
3% chance for additional ore from ore nodes
+1% luck

So in my examples:
- Maxed out combat would gain +40% damage, +20% crit chance, +40% HP. 
- Maxed out fishing with be +40% Catch Speed, +40% Fishing Bar Size, + 100% Fish Sell Price. 
- Maxed out farming would be +20% chance of crops being highest quality + 40% sell prices.
- Etc etc.

I think this would be pretty cool.

**Comment by ForGodsSakesValatros** [comment]
I definitely love the idea of integrating it with masteries in some way; even a simpler change of having the existing system, but requiring accrual of mastery instead of resetting would be a solid update for 1.6. Though there is something to be said for having to "sacrifice" to gain more under the old system, which would be missing. A complicated integration to be sure, looking forward to it. Masteries are neat but the game just feels lesser without MARGO...

**Comment by Biqnary** [comment]
I thought it would be cool if you needed to have ''maxed'' all skills and gotten all the professions to get the mastery bonuses, maybe each time you gain a new lvl 10 proffession you get some smaller buff and when you get all 4 you are able to get the mastery bonus for like a prismatic shard. also having the mastery bonuses be a bit... larger, they are fine for vanilla but with Margo the current bonuses feel lackluster.

**Comment by daleao** [comment]
If I understood correctly, your pitch is to replace both the skill reset and prestige mechanics.
Skill resets would be replaced by additional 3 mastery levels.
The Prestige profession choice would be removed entirely, replaced with another 5 mastery levels which reward the same flat stat bonuses to everyone.

Unfortunately this goes against my design philosophy.
The skill reset mechanic is meant to offset the (incredibly OP) ability to gain new professions by needing to overcome the challenge of relinquishing your skill levels and re-leveling from scratch.
This creates a two-dimensional progression system (can move up and down).
Moreover, the choice of _when_ to perform the reset becomes a mechanic in itself. Maybe you should wait until the next harvest, or until a certain level in the Mines.
My philosophy is always to create opportunities for meaningful gameplay choices, because to me _that's_ interesting.

In your proposal, the progression is one-dimensional and doesn't offer any choices.
It's basically "line always go up", which I find much less interesting.



lso my design intent to force players to make choices.
3.  is offset by needing to re-level the skill from scratch. Effectively removes 2 distinct mechanicswhich makes the entire process of mastering your skills less interesting 
with extra points in the vanilla mastery mechanic, and then replace the one-time pr with even more mastery points t Sounds like you are proposing replacing skill reset mechanic with the prestige profession variants with 5 extra levels of skill-specific stat buffs?
If so, I feel like these stat buffs diminish the value of gear and accessories.

**Comment by argon932** [comment]
No worries daleao. Thanks for the reply and I am excited to see what you come up with for 1.6.

**Comment by KariArisu** [comment]
To me it's the opposite, resetting profession levels was the worst part of the entire process. It wasn't interesting imo, and thus I always played with extra exp on resets to make it less annoying. My goal is to have all of the professions at the cost of extra grind, and having some professions MUCH later than others. If anything, going through mastery at least means you have to hit 10 on every skill instead of being able to hyper-focus on one skill without touching the rest.

Either way, mastery feels like a good opportunity for this mod regardless of what you end up doing with it. I like extending the grind for big rewards while modding the rest of the game to be harder.

**Comment by argon932** [comment]
> To me it's the opposite, resetting profession levels was the worst part of the entire process. It wasn't interesting imo, and thus I always played with extra exp on resets to make it less annoying. My goal is to have all of the professions at the cost of extra grind, and having some professions MUCH later than others. If anything, going through mastery at least means you have to hit 10 on every skill instead of being able to hyper-focus on one skill without touching the rest.
> 
> Either way, mastery feels like a good opportunity for this mod regardless of what you end up doing with it. I like extending the grind for big rewards while modding the rest of the game to be harder.

I agree with you but I understand why some people like that mechanic. I personally just like grinding more and always going upward. My favorite feature of MARGO was being able to get all profession perks, extending my gametime, but every time I prestige I died a little inside with the feeling of "Oh boy, here we go again". I disabled unlearning of recipes, which helps, but I still have never liked the concept of unlearning something to learn more of it. Prestiging in games has never made logical sense to me - but again - I see why some people enjoy the mechanic. Can only hope the update will have a system to utilize mastery in a way that can please both groups of people.

**Comment by Saturday13th** [comment]
So when will there be the next update for 1.6?

**Comment by Looneycore** [comment]
Adding on to some of the thoughts above, I have a thought for a way to integrate mastery and prestige. A little similar to [Biqnary](https://github.com/Biqnary)'s idea. My idea is to integrate prestige with the mastery cave, rather than the sewer statue. Once a player reaches 10 in ALL skills, they can go to the mastery cave and interact with "something" (maybe the final path pedestal in the middle) that will reset ALL of your skills to 0 and grant 1 mastery point. On the 4th interact instead of resetting your skills, this would give you a mastery point and then unlock leveling your skills to 20. Interact when all skills are 20 to then receive the final mastery point. I know this takes away the player agency in choosing to reset each skill individually, but I feel this may integrate prestige and mastery in a meaningful way. This would also allow for the sewer statue to be used in early game if you wanted to swap professions before having prestiged.

To make it a little clearer:
Mastery 1 = First prestige,  1 Profession
Mastery 2 = Second prestige, 2 Profession
Mastery 3 = Third prestige, 3 Profession
Mastery 4 = All skills at 10; Unlocks leveling to 20; Will have each profession at this point.
Mastery 5 = All skills at 20;

A benefit of this will be that a player needs to be more focused on all aspects of the game, rather than easily getting farming to 20 before other skills have been prestiged even once.

I don't know if you will implement any of this, just thought this is how I would implement it and wanted to share my thoughts. 

Sending you good vibes as you update MARGO to be 1.6 Compatible!
